### ISSUE TITLE
Update operations fail for Boolean fields

### ISSUE THREAD

**Issue by jpthomson** [vespa-engine/vespa#9452]
Switching from `put` to `update` operations we're getting a consistent error when trying to update the value of a boolean field.

Given a bool field:

```
    field public type bool {
      indexing: summary | attribute
    }
```

And an update operation:

```
{"update":"id:fy:product::unknown-portrait-disaster-7-art-print",
 "create":true,
 "fields":{"public":{"assign":"true"}}}
```

We see the following error using the Java feeding library:
```
11:39:40.006 search-update staging.log [INFO] [2019-05-20 09:39:40,007] [fy.search-update.system] [async-dispatch-3] - {:id id:fy:product::unknown-portrait-disaster-7-art-print, :result {:document-id id:fy:product::unknown-portrait-disaster-7-art-print, :operation-id d0fsbkt67789c, :success false, :details [{:success false, :result-type FATAL_ERROR, :exception #error {
 :cause Processing failed. Error message: java.lang.IllegalArgumentException: Class class com.yahoo.document.datatypes.BoolFieldValue not applicable to an class com.yahoo.document.datatypes.BoolFieldValue instance. -- See Vespa log for details.
 :via
 [{:type java.lang.RuntimeException
   :message Processing failed. Error message: java.lang.IllegalArgumentException: Class class com.yahoo.document.datatypes.BoolFieldValue not applicable to an class com.yahoo.document.datatypes.BoolFieldValue instance. -- See Vespa log for details.
   :at [com.yahoo.vespa.http.client.core.operationProcessor.EndPointResultFactory parseResult EndPointResultFactory.java 79]}]
 :trace
 [[com.yahoo.vespa.http.client.core.operationProcessor.EndPointResultFactory parseResult EndPointResultFactory.java 79]
  [com.yahoo.vespa.http.client.core.operationProcessor.EndPointResultFactory createResult EndPointResultFactory.java 35]
  [com.yahoo.vespa.http.client.core.communication.IOThread processResponse IOThread.java 251]
  [com.yahoo.vespa.http.client.core.communication.IOThread feedDocumentAndProcessResults IOThread.java 269]
  [com.yahoo.vespa.http.client.core.communication.IOThread pullAndProcessData IOThread.java 290]
  [com.yahoo.vespa.http.client.core.communication.IOThread cycle IOThread.java 340]
  [com.yahoo.vespa.http.client.core.communication.IOThread run IOThread.java 378]
  [java.lang.Thread run Thread.java 834]]}, :trace-message }]}}
```

And the following in the vespa logs:
```
[2019-05-20 09:39:40.424] WARNING : container        Container.com.yahoo.docproc.jdisc.DocumentProcessingTask    Processing of update of document 'id:fy:product::unknown-portrait-disaster-7-art-print': create-if-non-existent=true: 'public' [assign true]]. Last call: call to class com.yahoo.docprocs.indexing.IndexingProcessor (id: com.yahoo.docprocs.indexing.IndexingProcessor in indexing) failed at call to class com.yahoo.docprocs.indexing.IndexingProcessor (id: com.yahoo.docprocs.indexing.IndexingProcessor in indexing)\nexception=\njava.lang.IllegalArgumentException: Class class com.yahoo.document.datatypes.BoolFieldValue not applicable to an class com.yahoo.document.datatypes.BoolFieldValue instance.\n\tat com.yahoo.document.datatypes.BoolFieldValue.assign(BoolFieldValue.java:56)\n\tat com.yahoo.vespa.indexinglanguage.FieldUpdateHelper.createFieldValue(FieldUpdateHelper.java:66)\n\tat com.yahoo.vespa.indexinglanguage.FieldUpdateHelper.applyUpdate(FieldUpdateHelper.java:52)\n\tat com.yahoo.vespa.indexinglanguage.SimpleAdapterFactory.newUpdateAdapterList(SimpleAdapterFactory.java:69)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:41)\n\tat com.yahoo.docprocs.indexing.DocumentScript.execute(DocumentScript.java:66)\n\tat com.yahoo.docprocs.indexing.IndexingProcessor.processUpdate(IndexingProcessor.java:107)\n\tat com.yahoo.docprocs.indexing.IndexingProcessor.process(IndexingProcessor.java:64)\n\tat com.yahoo.docproc.Call.call(Call.java:160)\n\tat com.yahoo.docproc.DocprocExecutor.process(DocprocExecutor.java:120)\n\tat com.yahoo.docproc.jdisc.DocumentProcessingTask.process(DocumentProcessingTask.java:131)\n\tat com.yahoo.docproc.jdisc.DocumentProcessingTask.run(DocumentProcessingTask.java:65)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\n
```

At first glance it looks like BoolFieldValue may be missing one or more coercions

https://github.com/vespa-engine/vespa/blob/6f4f7be491b9f29875c81cbb8c7240307372a0b0/document/src/main/java/com/yahoo/document/datatypes/BoolFieldValue.java#L50-L58

**Comment by baldersheim** [comment]
Not good. Which vespa version ?

**Comment by jpthomson** [comment]
We're on `7.34.16` however I see the same behaviour running `7.45.22` locally.

**Comment by baldersheim** [comment]
This should be fixed with #9462

**Comment by baldersheim** [comment]
This should be fixed in the upcoming 7.50.x release in a day or two.
